Adjust e2e browser based tests

garciafdezpatricia opened the pull request:
# New feature description
The Dev-2-3 environment introduces a restriction on the clients that can issue an Access Grant, breaking e2e tests whose target environment is Dev-2-3 since they're not acting as an allowed client. 
The adjustment replaces the use of access grants with access requests, which don't have any restriction and don't need AMC to be involved in the process: instead of approving and revoking an access grant, an access request is issued and denied.
The test-app UI has also been modified according to the changes.

# Checklist

- [ ] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

garciafdezpatricia committed:
Adjust e2e browser based tests

The Dev-2-3 environment introduces a restriction on the clients that can issue an Access Grant. This breaks e2e tests since they're not an allowed client.
The adjustment replaces the use of access grants with access requests, which don't have any restriction and don't need AMC to be involved in the process.

garciafdezpatricia committed:
Adjust e2e browser based tests

The Dev-2-3 environment introduces a restriction on the clients that can issue an Access Grant. This breaks e2e tests since they're not an allowed client.
The adjustment replaces the use of access grants with access requests, which don't have any restriction and don't need AMC to be involved in the process.

NSeydoux commented on e2e/browser/test-app/components/accessGrants/index.tsx:
```diff
@@ -39,13 +42,13 @@ import { useRouter } from "next/router";
 }) {
   const [accessGrant, setAccessGrant] = useState<AccessGrant>();
-  const [accessRequest, setAccessRequest] = useState<string>();
+  const [accessRequest, setAccessRequest] = useState<AccessRequest | string>();
```
Let's not have an ambiguous type like this, and either keep it a plain string or an `AccessRequest` object.

NSeydoux commented on e2e/browser/test-app/components/accessGrants/index.tsx:
```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
-    );
-    setAccessGrant(accessGrantRequest);
+    // if the resource owner is not defined, do nothing.
+    if (session.info.webId) {
```
This test could be added to the clause of line 97. It's usually more readable to have an early return (and so to test for negative conditions) rather than testing for positive conditions and having the "main" code inside the conditional branch.

NSeydoux commented on e2e/browser/test-app/components/accessGrants/index.tsx:
```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
+    if (session.info.webId) {
+      const accessRequestReturned = await issueAccessRequest(
+        {
+          requestor: "https://johndoe.webid",
```
`requestor` is a deprecated parameter and should be removed.

NSeydoux commented on e2e/browser/test-app/components/accessGrants/index.tsx:
```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
+            "https://w3id.org/dpv#UserInterfacePersonalisation",
+            "https://w3id.org/dpv#OptimiseUserInterface",
+          ],
+          inherit: false,
```
If this isn't strictly required, let's not include it at all and use the defaults.

NSeydoux commented on e2e/browser/test-app/components/accessGrants/index.tsx:
```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
-    if (typeof accessGrant === "undefined") {
+  const handleDenial = async () => {
+    if (typeof accessRequest === "undefined") {
       // If the resource does not exist, do nothing.
       return;
     }
-    await revokeAccessGrant(accessGrant, {
+    await denyAccessRequest(accessRequest, {
       fetch: session.fetch,
     });
-    setAccessGrant(undefined);
+    setAccessRequest(undefined);
```
Denying the Access grant is a privileged operation that issues an Access Grant, it has to be performed by an authorized client. This should cancel the Access Request instead.

NSeydoux reviewed (changes requested):
I know cancelling the Access Request is currently blocked by a bug. Let's pause this PR for the time being, fix the bug, and then rework this once the fix has been merged.

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
cancelAccessRequest supports revoking an Access Request

garciafdezpatricia committed:
cancelAccessRequest supports revoking an Access Request

garciafdezpatricia committed:
review and tests fixes

Created an internal function to revoke an access credential (access grant or access request). This function is now used by cancelAccessRequest and revokeAccessGrant. Consequently adapted unit tests.

garciafdezpatricia committed:
review and tests fixes

Created an internal function to revoke an access credential (access grant or access request). This function is now used by cancelAccessRequest and revokeAccessGrant. Consequently adapted unit tests.

garciafdezpatricia committed:
Merge branch 'fix/SDK-3328/type-checks-access-grants' into chore/SDK-3325/access-grant-to-access-request-e2e-browser

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
review fixes

garciafdezpatricia committed:
Merge branch 'fix/SDK-3328/type-checks-access-grants' into chore/SDK-3325/access-grant-to-access-request-e2e-browser

garciafdezpatricia committed:
jest func coverage threshold

garciafdezpatricia committed:
jest func coverage threshold

garciafdezpatricia committed:
Merge branch 'chore/SDK-3325/access-grant-to-access-request-e2e-browser' of https://github.com/inrupt/solid-client-access-grants-js into chore/SDK-3325/access-grant-to-access-request-e2e-browser

edwardsph commented:
This has been incorporated into https://github.com/inrupt/solid-client-access-grants-js/pull/999
